src/tracking_state.jl

```diff
@@ -157,6 +202,62 @@ function TrackingState(
     )
 end
 
+# CUDA dispatch
+function TrackingState(
```

Comment:
This constructor might not be necessary any more, if you remove `DownconvertedSignalGPU` and `CarrierReplicaGPU`.

Reply:
Commenting after https://github.com/JuliaGNSS/Tracking.jl/pull/33/commits/ccb75f056f19bd0ddf504a4025bd0a304b11470d and https://github.com/JuliaGNSS/Tracking.jl/pull/33/commits/2a24f7d8dcdb95a468f32845070350263d6684a2: I thought that it would be quite strange to use `CarrierReplicaCPU` for a blank GPU carrier, hence I declared a function called `gen_blank_carrier_gpu` [here](https://github.com/JuliaGNSS/Tracking.jl/pull/33/commits/2a24f7d8dcdb95a468f32845070350263d6684a2#diff-0d80044177cc2ba8dce9f484cd531f4fc44523457df7fe25e82d88dd1c47228cR53). The CPU `TrackingState` generation differs from GPU `TrackingState` generation in the declaration of the variables `downconverted_signal`, `code`, and `carrier`. The first two are left as `Nothing`, as these are not fetched from the kernel, the latter is fed to the kernel as an empty `StructArray` for proper carrier generation. Do you think it's okay this way?